<?xml version="1.0" ?>
<launch>

	<!-- Kinect -->
	<include file="$(find azure_kinect_ros_driver)/launch/driver.launch">
		<arg name="depth_mode" value="NFOV_UNBINNED" />
		<arg name="color_enabled" value="false" />
		<arg name="fps" value="15" />
		<arg name="point_cloud" value="false" />
	</include>

    <!-- Spawn an image_proc/rectify nodelet to rectify the depth image -->
    <node pkg="nodelet" type="nodelet" name="manager" args="manager" output="screen">
      <param name="num_worker_threads" value="16" />
    </node>
    <node pkg="nodelet" type="nodelet" name="rectify_depth"
          args="load image_proc/rectify manager --no-bond"
          respawn="true">
      <remap from="image_mono"  to="depth/image_raw" />
      <remap from="image_rect"  to="depth/image_rect" />
      <param name="interpolation" value="0" />
    </node>
    <node pkg="nodelet" type="nodelet" name="k4a_ros_bridge"
          args="load Azure_Kinect_ROS_Driver/K4AROSBridgeNodelet manager --no-bond"
          respawn="true">
      <param name="depth_enabled"     value="true" />
      <param name="depth_mode"        value="NFOV_UNBINNED" />
      <param name="color_enabled"     value="false" />
      <param name="fps"               value="15" />
      <param name="point_cloud"       value="false" />
      <param name="required"          value="true" />
    </node>

	<!-- Main script-->
	<node name="camera_env" pkg="franka_irom" type="cameraGraspEnvDRPL.py" output="screen"/>

</launch>

